code/parachain/frame/cosmwasm/src/runtimes/abstraction.rs

```diff
@@ -97,7 +97,7 @@ pub enum GasOutcome {
 
 impl Gas {
 	pub fn new(max_frames: u32, initial_value: u64) -> Self {
-		let max_frames = usize::try_from(max_frames).unwrap();
+		let max_frames = max_frames as usize;
```

Comment:
But why?  `as` is a tricky way to cast since it may loose precision.  With `try_from` we’re sure we’ll never loose data and on platforms where usize is 32-bit or 64-bit wide both compile to the same code.

Reply:
max_frames is u32. so you saying for case when usize is u16? or we going to increase u32 to u64? but in this case would not it mean usize sure is 16. also why we should potentially brick and panic runtime, while loosing precision we just execute without unwrap linter? we have unwrap linter, i am fixing. what is best fix? unwraps sometimes bad sometime good. but would you prefer suppress linter? or just execute?

i prefer exactly `as usize` code as it will just work without linter noise and susipicous unwrap

Reply:
I’m saying, always prefer `try_from` over `as` so that the code is consistent everywhere and we don’t silently loose precision.  In other places we may have `u64` argument cast to `usize` and there the check is necessary.

> also why we should potentially brick and panic runtime

Because if the value doesn’t fit the type we’re casting to, that’s a bug in code and the best thing to do is abort the program.

> unwraps sometimes bad sometime good

Unwraps are bad for input validation or error handling.  They are good in situations where triggering the unwrap would indicate a bug in the code.  E.g. in situations like her where programmer just assumes that they can cast `max_frames` to `usize`.

I find the lint check counterproductive.  `unwrap` ends up being replaced by `expect` with a meaningless message or by a worse code.  Furthermore, it has too many false negatives.  Indexing for example is essentially a hidden unwrap call.

Reply:
unwrap must not happen in parachains. never. if you know that it never happens - please write down exactly why - expect. or loose precision. I prefer loose precision than paying parity 20K USD to unbrick chain.

Reply:
this is default guide line for Substrate and linter in ecosystem. btw in CW there is no stacktrackes, so without expect you will not see anything at all. what really failed.

so not sure why you are so on unwrap when it is terrible...

Reply:
> Furthermore, it has too many false negatives. Indexing for example is essentially a hidden unwrap call.

We have linter forbidding indexing, you are right. We have it.

Reply:
> expect with a meaningless message or by a worse code

why it is meaningless? for me it at least something. i wrote before about it.

Reply:
btw in this case real fix is to use usize or u16 for frame size. so we do not even have option to panic here or do unwraps.